ElectricCar.cpp

```diff
@@ -1,16 +1,50 @@
 #include "ElectricCar.hpp"
+
 #include <iostream>
 
-ElectricCar::ElectricCar(ElectricEngine* engine)
-    : engine_(engine)
-{
+#include "InvalidEngineChange.hpp"
+
+ElectricCar::ElectricCar(ElectricEngine* electricEngine)
+    : electricEngine_(electricEngine) {
     std::cout << __FUNCTION__ << std::endl;
 }
 
-ElectricCar::~ElectricCar()       { std::cout << __FUNCTION__ << std::endl; }
-void ElectricCar::turnLeft()      { std::cout << __FUNCTION__ << std::endl; }
-void ElectricCar::turnRight()     { std::cout << __FUNCTION__ << std::endl; }
-void ElectricCar::brake()         { std::cout << __FUNCTION__ << std::endl; }
-void ElectricCar::accelerate(int) { std::cout << __FUNCTION__ << std::endl; }
-void ElectricCar::charge()        { std::cout << __FUNCTION__ << std::endl; }
+ElectricCar::~ElectricCar() {
+    delete electricEngine_;
+    std::cout << __FUNCTION__ << std::endl;
+}
+void ElectricCar::charge() {
+    std::cout << __FUNCTION__ << std::endl;
+}
+
+void ElectricCar::restore() {
+    charge();
+}
 
+uint16_t ElectricCar::getBatteryCapacity() const {
+    return electricEngine_->getBatteryCapacity();
+}
+
+void ElectricCar::carInfo() {
+    std::cout << "Car info: \n "
+              << "Speed: " << getSpeed() << '\n'
+              << "Power: " << getEnginePower() << '\n'
+              << "Battery Capacity: " << getBatteryCapacity() << '\n';
+}
+
+uint16_t ElectricCar::getEnginePower() const {
+    return electricEngine_->getPower();
+}
+
+void ElectricCar::changeEngine(Engine* engine) {
+    if (getSpeed() != MINIMUM_VELOCITY) {
```

Comment:
unsafe operation

Reply:
Could you explain what is unsafe there? What should i change?

Reply:
Now it's safe since you are comparing int with int.

Reply:
Ohh Ok. Thanks @WBigaj ;~}